iceoryx2-services/discovery/src/service_discovery/service.rs

```diff
@@ -304,11 +359,25 @@ impl<S: ServiceType> Service<S> {
             tracker.sync(iceoryx_config)?;
         }
 
+        let server = Some(
```

Comment:
You should keep the service object here, otherwise the service is created with the server and then, if the client does not connect on time the service is cleaned up in that sense that it is no longer discoverable.

Reply:
@RTK-jangid Did you miss this one or are you choosing to not address it ? If the latter, could you explain your reasoning?

In case there is some confusion, this finding is suggesting to keep the **service** object alive in addition to the **server** object.